cli/tools/upgrade.rs

```diff
@@ -842,6 +890,17 @@ mod test {
     fn current_version(&self) -> Cow<str> {
       Cow::Owned(self.current_version.borrow().clone())
     }
+
+    // TODO(bartlomieju): update to handle `Lts` and `Rc` channels
+    async fn get_current_exe_release_channel(
+      &self,
+    ) -> Result<ReleaseChannel, AnyError> {
+      if *self.is_canary.borrow() {
+        Ok(ReleaseChannel::Canary)
+      } else {
+        Ok(ReleaseChannel::Stable)
+      }
+    }
```

Comment:
It does feel like the strategy pattern would be better for all this.

Reply:
I agree, but I want to untangle more stuff in this file before doing this.